Assets/Scripts/UI/TwitchLeaderboard.cs

```diff
@@ -31,50 +27,74 @@ private void Start()
             return;
         }
 
-        float delay = 0.6f;
-        int index = 0;
+        if (leaderboard.Success)
+        {
+            promptTransform.FindChild("Retry").gameObject.SetActive(false);
+        }
 
-        foreach (Leaderboard.LeaderboardEntry entry in leaderboard.GetSortedEntries(maximumRowCount))
+        statsTable = Instantiate<TwitchLeaderboardStats>(twitchLeaderboardStatsPrefab);
+        statsTable.leaderboard = leaderboard;
+        
+        if (leaderboard.Count > 0)
         {
-            TwitchLeaderboardRow row = Instantiate<TwitchLeaderboardRow>(index < specialRows.Length ? specialRows[index] : normalRow);
-            row.position = index + 1;
-            row.leaderboardEntry = entry;
-            row.delay = delay;
-            row.transform.SetParent(tableTransform, false);
+            mainTable = Instantiate<TwitchLeaderboardTable>(twitchLeaderboardTablePrefab);
+            mainTable.leaderboard = leaderboard;
+        }
 
-            _instancedRows.Add(row);
+        if (leaderboard.SoloCount > 0)
+        {
+            soloTable = Instantiate<TwitchLeaderboardTableSolo>(twitchLeaderboardTableSoloPrefab);
+            soloTable.leaderboard = leaderboard;
+            transform.FindChild("UI").FindChild("Mask").gameObject.SetActive(true);
```

Comment:
The "mask" is a black background and gradient on the left side of the prefab (TwitchLeaderboard/UI/Mask), which is just there to cover over the binder when there's content on the left side. The reason for activating it in code is that, if there have been no solo clears and less than 16 players, the leaderboard uses the same right-side-only layout as the current TP version.

I'm a bit uncomfortable with this, as I'm pretty sure using a black box isn't the right way to cover something up. I don't have the knowledge of Unity currently to do this in a more appropriate manner, so would appreciate some input from someone with more experience.